server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -7567,6 +7570,9 @@ public void onIndexCommitDelete(ShardId shardId, IndexCommit deletedCommit) {
             }
 
             releaseCommitRef(acquiredCommits, 7L);
+
+            final long primaryTerm = engine.config().getPrimaryTermSupplier().getAsLong();
+            assertThat(acquiredPrimaryTerms, hasItem(primaryTerm));
```

Comment:
Would we not expect all primary terms to be the same, rather than only seeing it once?

Reply:
Yes, thanks for catching this, I pushed 6b6147815993c6e60b837dba740d72b88eb73f33